docs/workflow/debugging/coreclr/debugging-runtime.md

```diff
@@ -0,0 +1,223 @@
+
+Steps 1-9 only need to be done once as long as there's been no changes to the CMake files in the repository. Afterwards, step 10 can be repeated whenever you want to start debugging. As of now, it is highly recommended to use Visual Studio 2022 or Visual Studio 2019.
+
+#### Using Visual Studio Open Folder with CMake
+
+1. Open the _dotnet/runtime_ repository in Visual Studio using the _open folder_ feature. When opening the repository root, Visual Studio will prompt about finding _CMake_ files. Select `src\coreclr\CMakeList.txt` as the CMake workspace.
+2. Set the `corerun` project as startup project. When using the folder view instead of the CMake targets view, right click `coreclr\hosts\corerun\CMakeLists.txt` to set as startup project, or select it from debug target dropdown of Visual Studio.
+3. Right click the `corerun` project and open the `Debug` configuration. You can also click on _Debug -> Debug and Launch Configuration_ from the Visual Studio main bar menu.
+4. In the opened `launch.vs.json`, set following properties to the configuration of `corerun`:
+
+```json
+    {
+      "type": "default",
+      "project": "CMakeLists.txt",
+      "projectTarget": "corerun.exe (hosts\\corerun\\corerun.exe)",
+      "name": "corerun.exe (hosts\\corerun\\corerun.exe)",
+      "environment": [
+        {
+          "name": "CORE_ROOT",
+          "value": "${cmake.installRoot}"
+        },
+        {
+          "name": "CORE_LIBRARIES",
+          // for example net7.0-windows-Debug-x64
+          "value": "${cmake.installRoot}\\..\\..\\runtime\\<tfm>-windows-<configuration>-<arch>\\"
+        }
+      ],
+      "args": [
+        // path to a managed application to debug
+        // remember to use double backslashes (\\)
+        "HelloWorld.dll"
+      ]
+    }
+```
+
+**NOTE**: For Visual Studio 17.3, changing the location of launched executable doesn't work, so the `CORE_ROOT` is necessary.
+
+5. Right click the CoreCLR project or `coreclr\CMakeLists.txt` in the folder view, and then invoke the _Install_ command.
+6. Press F10 or F11 to start debugging at main, or set a breakpoint and press F5.
+
+Whenever you make changes to the CoreCLR source code, don't forget to invoke the _Install_ command again to have them set in place.
+
+### Using Visual Studio Code
+
+It will be very nice to be able to achieve all this using Visual Studio Code as well, since it's the editor of choice for lots of developers.
+
+Visual Studio Code instructions coming soon!
+
+### Using SOS with Windbg or Cdb on Windows
+
+Under normal circumstances, SOS usually comes shipped with Windbg, so no additional installation is required. However, if this is not the case for you, you want to use another version, or any other circumstance that requires you to install it separately/additionally, here are two links with useful information on how to get it set up:
+
+* The official [Microsoft docs on SOS](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-sos).
```

Comment:
Please remove `en-us` from all URLs. This forces localization instead of letting the users browser provide the settings.